oauthproxy.go

```diff
@@ -838,22 +839,37 @@ func (p *OAuthProxy) enrichSessionState(ctx context.Context, s *sessionsapi.Sess
 func (p *OAuthProxy) AuthOnly(rw http.ResponseWriter, req *http.Request) {
 	session, err := p.getAuthenticatedSession(rw, req)
 	if err != nil {
-		http.Error(rw, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
-		return
-	}
+		if authOnlyRedirectSignIn(req) {
+			logger.Printf("Initiating AuthOnly SignIn Redirect")
+			if isAjax(req) {
+				// no point redirecting an AJAX request
+				p.errorJSON(rw, http.StatusUnauthorized)
+				return
+			}
 
-	// Unauthorized cases need to return 403 to prevent infinite redirects with
-	// subrequest architectures
-	if !authOnlyAuthorize(req, session) {
-		http.Error(rw, http.StatusText(http.StatusForbidden), http.StatusForbidden)
-		return
-	}
+			if p.SkipProviderButton {
+				p.OAuthStart(rw, req)
+			} else {
+				p.SignInPage(rw, req, http.StatusForbidden)
+			}
```

Comment:
What would happen if Traefik received a 302 code in the response here? Could it follow the redirect?

Reply:
I have looked again at Traefik ForwardAuth middleware docs, and seems to indicate that it will `return the response from the authentication server to the client`, which perhaps indicates that a 302 redirect might work. I'll give it a try and get back to you on that one

Reply:
I have added a redirect option now and left the commented out code for rendering options as well, interested to hear your thoughts